modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1124,7 +1124,10 @@ protected CompletableFuture<IgniteBiTuple<ClusterNode, Long>> enlist(int partId,
         // TODO asch IGNITE-15091 fixme need to map to the same leaseholder.
         // TODO asch a leader race is possible when enlisting different keys from the same partition.
         return fut0.handle((primaryPeerAndTerm, e) -> {
-            if (primaryPeerAndTerm.get1() == null || e != null) {
+            if (e != null) {
+                throw new TransactionException("Failed to get the primary replica.", e);
```

Comment:
I would rather use withCause with REPLICA_UNAVAILABLE_ERR code here.

Reply:
Thanks, I did not find that error code myself. Fixed

Reply:
I hope I got this right this time :)